src/SonarScanner.MSBuild.PreProcessor/AnalysisConfigProcessing/Processors/AnalysisConfigProcessorBase.cs

```diff
@@ -0,0 +1,50 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using SonarScanner.MSBuild.Common;
+
+namespace SonarScanner.MSBuild.PreProcessor.AnalysisConfigProcessing.Processors;
+
+public abstract class AnalysisConfigProcessorBase(ProcessedArgs localSettings, IDictionary<string, string> serverProperties) : IAnalysisConfigProcessor
```

Comment:
When we compile for net 4.6.2, the compiler will generate the actual constructor so this should not cause any issues when running on dotnet framework:
```csharp
protected AnalysisConfigProcessorBase(
      ProcessedArgs localSettings,
      IDictionary<string, string> serverProperties)
    {
      this.LocalSettings = localSettings;
      this.ServerProperties = serverProperties;
      // ISSUE: explicit constructor call
      base.\u002Ector();
    }
```